state/pruner.go

```diff
@@ -256,6 +278,7 @@ func (p *Pruner) pruneBlocksToRetainHeight(lastRetainHeight int64) int64 {
 	if err != nil {
 		p.logger.Error("Failed to prune blocks", "err", err, "targetRetainHeight", targetRetainHeight, "newRetainHeight", newRetainHeight)
 	} else if pruned > 0 {
+		p.metrics.BlockStoreBaseHeight.Set(float64(newRetainHeight))
```

Comment:
Is it possible for `pruned` to be 0, but the new retain height to change?

Reply:
No, but I want to set the metric only if there is a change. Otherwise there is no point in setting the metric.